tests/test_gp_interp.py

```diff
@@ -13,29 +13,37 @@
 #    and/or other materials provided with the distribution.
 
 from __future__ import print_function
-import warnings
 import galsim
+import treegp
 import numpy as np
 import piff
-import os
-import subprocess
-import yaml
-import fitsio
+from sklearn.model_selection import train_test_split
 
 from piff_test_helper import get_script_name, timer
 
-fiducial_kolmogorov = galsim.Kolmogorov(half_light_radius=1.0)
-mod = piff.GSObjectModel(fiducial_kolmogorov, centered=False, include_pixel=False,
-                         fastfit=True)
+kolmogorov = galsim.Kolmogorov(half_light_radius=1., flux=1.)
 
-star_type = np.dtype([('u', float),
-                      ('v', float),
-                      ('hlr', float),
-                      ('g1', float),
-                      ('g2', float),
-                      ('u0', float),
-                      ('v0', float),
-                      ('flux', float)])
+def return_var_map(weight, xi):
+    """
+    function to return variance of 2-pcf.
+
+    :param weight: weight, output of treegp.
+    :param xi:     2-pcf, ouput of treegp.
+    """
+    N = int(np.sqrt(len(xi)))
+    var = np.diag(np.linalg.inv(weight))
+    VAR = np.zeros(N*N)
+    I = 0
+    for i in range(N*N):
+        if xi[i] !=0:
+            VAR[i] = var[I]
+            I+=1
+        if I == len(var):
+            break
+    VAR = VAR.reshape(N,N) + np.flipud(np.fliplr(VAR.reshape(N,N)))
+    if N%2 == 1:
+        VAR[N/2, N/2] /= 2.
+    return VAR
```

Comment:
I'm pretty confused about this function.  What is the idea here?  Maybe could add some comments saying what the weight is that is output from treegp and what you actually want here as a variance estimate.  The math you're doing here is not at all obvious to me.

Especially please explain the bit where you test `xi[i] != 0`.  When is that expected?  Why would that test be expected to get you the right `I` indexing?

Reply:
I wrote this function for the pull map of the 2d fit. However I don’t need it here, so I removed it.